lib/phoenix_live_view/helpers.ex

```diff
@@ -826,4 +868,109 @@ defmodule Phoenix.LiveView.Helpers do
     raise ArgumentError,
           "live_title_tag/2 expects a :prefix and/or :suffix option, got: #{inspect(opts)}"
   end
+
+  @doc """
+  Renders a form function component using `Phoenix.HTML.form_for/4`.
+
+  ## Options
+
+  The `:for` assign is the form's source data and the optional `:action`
+  assign can be provided for the form's action. Additionally accepts
+  the same options as `Phoenix.HTML.form_for/4` as optional assigns:
+
+    * `:as` - the server side parameter in which all params for this
+      form will be collected (i.e. `as: :user_params` would mean all fields
+      for this form will be accessed as `conn.params.user_params` server
+      side). Automatically inflected when a changeset is given.
+
+    * `:method` - the HTTP method. If the method is not "get" nor "post",
+      an input tag with name `_method` is generated along-side the form tag.
+      Defaults to "post".
+
+    * `:multipart` - when true, sets enctype to "multipart/form-data".
+      Required when uploading files
+
+    * `:csrf_token` - for "post" requests, the form tag will automatically
+      include an input tag with name `_csrf_token`. When set to false, this
+      is disabled
+
+    * `:errors` - use this to manually pass a keyword list of errors to the form
+      (for example from `conn.assigns[:errors]`). This option is only used when a
+      connection is used as the form source and it will make the errors available
+      under `f.errors`
+
+    * `:id` - the ID of the form attribute. If an ID is given, all form inputs
+      will also be prefixed by the given ID
+
+  ## Examples
+
+      <.form let={f} for={@changeset}>
+        <%= text_input f, :name %>
+      </.form>
+
+      <.form let={user_form} for={@changeset} as="user" {@extra}>
+        <%= text_input user_form, :name %>
+      </.form>
+  """
+  @form_opts [:as, :method, :multipart, :csrf_token, :errors, :id]
+  def form(assigns) do
+    action = assigns[:action] || "#"
+
+    {form_for, assigns} =
+      Map.pop_lazy(assigns, :for, fn -> raise ArgumentError, "missing :for assign to form" end)
+
+    {csrf_token, assigns} =
+      Map.pop_lazy(assigns, :csrf_token, fn -> Phoenix.HTML.Tag.csrf_token_value(action) end)
+
+    form_opts = assigns |> Map.take(@form_opts) |> Enum.into([])
+
+    form = %Phoenix.HTML.Form{Phoenix.HTML.FormData.to_form(form_for, form_opts) | action: action}
```

Comment:
Ok, we definitely need to pass all options because they are stored under `form.options`. So I would go with something like this:


```suggestion
    {action, assigns} = Map.pop_lazy(assigns, :action, "#")

    {form_for, assigns} =
      Map.pop_lazy(assigns, :for, fn -> raise ArgumentError, "missing :for assign to form" end)

    form_opts = assigns_to_attributes(assigns)
    form = %Phoenix.HTML.Form{Phoenix.HTML.FormData.to_form(form_for, form_opts) | action: action}
```

For CSRF token, you can use assign_new later on.

    assigns =
      LiveView.assign_new(assigns, :csrf_token, fn -> Phoenix.HTML.Tag.csrf_token_value(action) end)